pkg/services/baremetal/host/host.go

```diff
@@ -1021,7 +1021,15 @@ func (s *Service) createAutoSetupInput(sshClient sshclient.Client) (autoSetupInp
 	if needsDownload {
 		out := sshClient.DownloadImage(imagePath, image.URL)
 		if err := handleSSHError(out); err != nil {
-			return autoSetupInput{}, actionError{err: fmt.Errorf("failed to download image: %s %s %w", out.StdOut, out.StdErr, err)}
+			err := fmt.Errorf("failed to download image: %s %s %w", out.StdOut, out.StdErr, err)
+			conditions.MarkFalse(
+				s.scope.HetznerBareMetalHost,
+				infrav1.ProvisionSucceededCondition,
+				infrav1.ImageSpecInvalidReason,
```

Comment:
this reason probably is not very precise looking at the function "downloadImage". Can you maybe list the reasons you know of for this to fail? Does it maybe even make sense to analyze the error?

Reply:
I created ImageDownloadFailedReason for this condition.